### ISSUE TITLE
Confusing `--help` description

### ISSUE THREAD

**Issue by akaihola** [gpt-engineer-org/gpt-engineer#1072]
## Expected Behavior

When running `gpte --help`, the output should explain the purpose of the software between the `Usage:` and `Arguments:` lines.

## Current Behavior

Instead, I get the docstring for the `main()` function, which is confusing:
```
$ gpte --help
```
```
Usage: gpte [OPTIONS] [PROJECT_PATH] [MODEL]

  The main entry point for the CLI tool that generates or improves a project.

  This function sets up the CLI tool, loads environment variables, initializes
  the AI, and processes the user's request to generate or improve a project
  based on the provided arguments.

  Parameters ---------- project_path : str     The file path to the project
  directory. model : str     The model ID string for the AI. temperature :
  float     The temperature setting for the AI's responses. improve_mode :
  bool     Flag indicating whether to improve an existing project. lite_mode :
  bool     Flag indicating whether to run in lite mode. clarify_mode : bool
  Flag indicating whether to discuss specifications with AI before
  implementation. self_heal_mode : bool     Flag indicating whether to enable
  self-healing mode. azure_endpoint : str     The endpoint for Azure OpenAI
  services. use_custom_preprompts : bool     Flag indicating whether to use
  custom preprompts. verbose : bool     Flag indicating whether to enable
  verbose logging.

  Returns ------- None

Arguments:
[...]
```

## The fix

This is trivial to fix:
```diff
diff --git a/gpt_engineer/applications/cli/main.py b/gpt_engineer/applications/cli/main.py
index 5d1a7cf..93ce8a0 100644
--- a/gpt_engineer/applications/cli/main.py
+++ b/gpt_engineer/applications/cli/main.py
@@ -137,7 +137,16 @@ def prompt_yesno(question: str) -> bool:
     return answer in ["y", "yes"]
 
 
-@app.command()
+@app.command(
+    help="""
+        GPT-engineer lets you:
+
+        \b
+        - Specify a software in natural language
+        - Sit back and watch as an AI writes and executes the code
+        - Ask the AI to implement improvements
+    """
+)
 def main(
     project_path: str = typer.Argument("projects/example", help="path"),
     model: str = typer.Argument("gpt-4-1106-preview", help="model id string"),
```

Typer/Click will ignore the eight-space indent for the help text, so we could format it like above for clarity.

The lonely `\b` prevents auto-filling paragraphs for the bullet points. See [click.wrap_text()](https://click.palletsprojects.com/en/8.1.x/api/#click.wrap_text) documentation.

Now the output looks better:
```
$ gpte --help
```
```
Usage: gpte [OPTIONS] [PROJECT_PATH] [MODEL]

  GPT-engineer lets you:

  - Specify a software in natural language
  - Sit back and watch as an AI writes and executes the code
  - Ask the AI to implement improvements

Arguments:
[...]
```
I can also put this in a pull request if it's not a too small contribution to be rejected.

**Comment by ATheorell** [comment]
It is a rather small PR, but you have researched it well. If you make a PR of this, I will merge it @akaihola

**Comment by ATheorell** [comment]
btw, could you add the fix from PR  #1024 to your PR?
